File: leetcode/problems/palindrome_number.rb
# Determine whether an integer is a palindrome. Do this without extra space.

# param {Integer} x
# return {Boolean}

def is_palindrome(x)
  return false if x < 0

  result, x_remaining = 0, x

  while x_remaining > 0
    result = result * 10 + x_remaining % 10
    x_remaining /= 10
  end

  result == x
end

# Time Complexity - O(n)
# Space Complexity - O(1)

File: leetcode/javascript_problems/palindrome_number.js
// Determine whether an integer is a palindrome.
// Do this without extra space.

/**
 * @param {number} x
 * @return {boolean}
 */

var isPalindrome = (x) => {
  let xStr = x.toString();

  let l = 0;
  let r = xStr.length - 1;

  while (l <= r) {
    if (xStr[l] !== xStr[r]) {
      return false;
    }

    l ++;
    r --;
  }

  return true;
};
